webpack-subresource-integrity throws error when a processed tag has no attributes

I have a project using webpack 5 and the latest versions of html-webpack-plugin and webpack-subresource-integrity. I also have a plugin that inlines scripts (e.g. webpack runtime chunk) into the generated HTML. This generates a `<script>` tag with no attributes, which causes webpack-subresource-integrity to throw when it checks the tag for an existing integrity attribute.

```
ERROR in   TypeError: Cannot convert undefined or null to object

  - hasOwnProperty

  - index.js:237 SubresourceIntegrityPlugin.processTag
    [client]/[webpack-subresource-integrity]/index.js:237:49

  - index.js:318
    [client]/[webpack-subresource-integrity]/index.js:318:48

  - Array.forEach

  - index.js:318
    [client]/[webpack-subresource-integrity]/index.js:318:26


  - new Promise


  - Hook.js:22 Hook.PROMISE_DELEGATE [as _promise]
    [client]/[html-webpack-plugin]/[tapable]/lib/Hook.js:22:14

  - index.js:311
    [client]/[html-webpack-plugin]/index.js:311:81



webpack 5.19.0 compiled with 1 error in 9396 ms
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! [project-name]@0.1.0 build:prod: `webpack --mode production`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the [project-name]@0.1.0 build:prod script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     E:\ProgramData\Cache\npm-cache\_logs\2021-01-31T02_37_05_041Z-debug.log

```

--- a/examples/issue-147/README.md
+++ b/examples/issue-147/README.md
@@ -1,0 +1,3 @@
+# Doesn't throw error when a processed tag has no attributes #hwp
+
+Test for https://github.com/waysact/webpack-subresource-integrity/issues/147

--- a/examples/issue-147/async.js
+++ b/examples/issue-147/async.js
@@ -1,0 +1,1 @@
+console.log("ok");

--- a/examples/issue-147/index.js
+++ b/examples/issue-147/index.js
@@ -1,0 +1,1 @@
+import("./async");

--- a/examples/issue-147/package.json
+++ b/examples/issue-147/package.json
@@ -1,0 +1,14 @@
+{
+  "name": "issue-147",
+  "description": "Test for https://github.com/waysact/webpack-subresource-integrity/issues/147",
+  "version": "1.0.0",
+  "license": "MIT",
+  "private": true,
+  "devDependencies": {
+    "html-webpack-plugin": "^5.0.0-beta.6",
+    "script-ext-html-webpack-plugin": "^2.1.5",
+    "webpack": "^5.19.0",
+    "webpack-cli": "*",
+    "webpack-subresource-integrity": "*"
+  }
+}

--- a/examples/issue-147/package.json
+++ b/examples/issue-147/package.json
@@ -6,6 +6,7 @@
   "private": true,
   "devDependencies": {
     "html-webpack-plugin": "^5.0.0-beta.6",
+    "nyc": "*",
     "script-ext-html-webpack-plugin": "^2.1.5",
     "webpack": "^5.19.0",
     "webpack-cli": "*",

--- a/examples/issue-147/webpack.config.js
+++ b/examples/issue-147/webpack.config.js
@@ -1,0 +1,33 @@
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const path = require("path");
+const ScriptExtHtmlWebpackPlugin = require("script-ext-html-webpack-plugin");
+const { SubresourceIntegrityPlugin } = require("webpack-subresource-integrity");
+
+module.exports = (env) => ({
+  entry: {
+    app: path.resolve(__dirname, "index"),
+  },
+  output: {
+    path: path.resolve(__dirname, "dist", "inline"),
+    crossOriginLoading: "anonymous",
+  },
+  optimization: {
+    runtimeChunk: {
+      // Put webpack runtime code in a single separate chunk called "runtime.js"
+      name: "runtime",
+    },
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      inject: "body",
+    }),
+    new ScriptExtHtmlWebpackPlugin({
+      inline: {
+        // Inline "runtime.js" as a <script> tag in the HTML
+        chunks: "initial",
+        test: "runtime",
+      },
+    }),
+    new SubresourceIntegrityPlugin(),
+  ],
+});

--- a/webpack-subresource-integrity/index.ts
+++ b/webpack-subresource-integrity/index.ts
@@ -419,7 +419,10 @@ export class SubresourceIntegrityPlugin {
    * @internal
    */
   private processTag = (compilation: Compilation, tag: HtmlTagObject): void => {
-    if (Object.prototype.hasOwnProperty.call(tag.attributes, "integrity")) {
+    if (
+      tag.attributes &&
+      Object.prototype.hasOwnProperty.call(tag.attributes, "integrity")
+    ) {
       return;
     }
 

